.github/workflows/deploy.yml

```diff
@@ -10,11 +10,11 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: gradle/wrapper-validation-action@v1
-      - uses: actions/setup-java@v2
+      - uses: actions/setup-java@v3
         with:
           java-version: 17
           distribution: 'temurin'
-      - uses: actions/cache@v2
+      - uses: actions/cache@v3
```

Comment:
Doesn't setup-java v3 include caching? Can we remove the actions/cache?

Reply:
Okay.
i delete the cache action in all the worksflows and set the cache in the setup java.

Reply:
there is some debate over the issue, but in my experience gradle's own [`gradle-build-action`](https://github.com/gradle/gradle-build-action) does the most effective caching on GHA.

Reply:
for me im not sure what can be better then i wait for another comment.

Reply:
If you want to change it to the gradle one, that's fine. But since most of our projects currently use the setup java caching, it's also ok if we switch to the gradle action in a separate PR later.

Reply:
Okay. i keep the changes and think is ready in a future PR this can be iterated again (?